programs/mango-v4/src/serum3_cpi.rs

```diff
@@ -155,6 +156,20 @@ impl OpenOrdersSlim {
             referrer_rebates_accrued: oo.referrer_rebates_accrued,
         }
     }
+    pub fn from_oo_v2(oo: &OpenOrdersV2, base_lot_size: u64, quote_lot_size: u64) -> Self {
+        let bids_base_lots: u64 = oo.position.bids_base_lots.try_into().unwrap();
+        let asks_base_lots: u64 = oo.position.asks_base_lots.try_into().unwrap();
+        let base_locked_native = bids_base_lots * base_lot_size;
+        let quote_locked_native = asks_base_lots * quote_lot_size;
```

Comment:
This sounds like it's the wrong side around: For bids we need to lock up quote and for asks we need to lock up base.

Also, bids_base_lots and asks_base_lots are both in _base_ units, meaning that to find the amount of quote locked up, we need to multiply with the price. Check out OpenOrdersAccount::cancel_order and use the same logic as there.

Reply:
(left this open because I want to double-check against the obv2 code)